src/pages/ProductList/PageButton/PageButton.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+import { API } from '../../../config';
+import './PageButton.scss';
+
+export class PageButton extends Component {
+  constructor() {
+    super();
+    this.state = {
+      navInfo: 0,
+      location: [],
+    };
+  }
+
+  componentDidMount() {
+    const { idInfo } = this.props;
+    fetch(API.NAVIGATOR)
+      .then(res => res.json())
+      .then(data => {
+        console.log('data:', data);
+        this.setState({
+          navInfo: data.navigators[idInfo.categoryId - 1],
+          location: idInfo,
+        });
+      });
+  }
+
+  render() {
+    const { pageHandler } = this.props;
+    const { categoryId, subcategoryId } = this.props.idInfo;
+    const { navInfo } = this.state;
+    let arrNum;
+    if (navInfo && !Number(subcategoryId)) {
+      (arrNum = []).length = Math.ceil(navInfo.products_count / 16);
+      arrNum.fill(0);
+    } else if (navInfo) {
+      (arrNum = []).length = Math.ceil(
+        navInfo.subcategories[Number(subcategoryId) - 1].products_count / 16
+      );
+      arrNum.fill(0);
+    }
```

Comment:
```suggestion
    const arrNum = [];
    if (navInfo && !Number(subcategoryId)) {
      (arrNum).length = Math.ceil(navInfo.products_count / 16);
      arrNum.fill(0);
    } else if (navInfo) {
      (arrNum).length = Math.ceil(
        navInfo.subcategories[Number(subcategoryId) - 1].products_count / 16
      );
      arrNum.fill(0);
    }
```
초기값을 배열로 주고 시작하는게 좋아보입니다.
그리고 arrNum은 재할당은 되지 않는 값이기에 const로 선언하는게 좀 더 알맞아보이네요

Reply:
arrNum이라는 변수명도 의미가 드러나도록 수정해주세요

Reply:
amountOfPage란 값으로 수정해서 초기값을 추가했습니다. 감사합니다!!